### DISCUSSION TITLE
@click.away not triggered when triggering another @click

### DISCUSSION THREAD

**Discussion by flore89** [alpinejs/alpine#3461] [1. Help]
Hi,
so I have a Megamenu toggle and a Search Modal toggle. They both have there own x-data and are not nested into each other.
Both elements close on @click.away. But if the Search Modal is open, and I click on the mega menu open trigger (@click). The search modal will no close, because the @click.away is not trigger when i click on the @click from the megamenu. 

Is this behaviour a bug or is it intended? How would i go about solving it?

**Comment by SimoTod** [comment]
Click.away triggers when the click event bubbles up to window. It should work even if there are other click directives involved as long as you don't stop the event.

**Comment by ekwoka** [comment]
Do you have any `@click.stop`?

**Comment by flore89** [comment]
no, there is no `@click.stop`

I prepared a codepen so it is easier to explain: https://codepen.io/flore89/pen/WNgzZPK

1. Click the search input
2. Search modal will open, and it closes on @click.away
3. Then click on any Megamenu
4. Megamenu will open, but the search modal is still visible because the click somehow doesn't count as @click.away
5. The same problem is vice versa. If Mega is open and search gets opened, it will not trigger @click.away

I hope my problem is clearer now.

Thank you!

**Reply by ekwoka** [reply]
![Screenshot 2023-03-14 at 5 17 29 PM](https://user-images.githubusercontent.com/43540491/225012883-82da61d2-06bd-41e8-bdc5-822b5e480fe7.png)

**Reply by ekwoka** [reply]
the `a` should be `button` btw

**Comment by flore89** [comment]
Damn, I'm sorry, was searching for `@click.stop` ... my bad! That did the trick. But now the megamenu doesn't work as intended.

Please see updated codepen: https://codepen.io/flore89/pen/WNgzZPK

When a megamenu is open, and I change to another one. It will not open anymore (on the first click). Guess that was why i had the `x-on:click.stop`

Do you have a solution where both behaviours are possible?

**Reply by ekwoka** [reply]
I would have the menus and menu buttons inside the same container with a @click.outside to close everything.
